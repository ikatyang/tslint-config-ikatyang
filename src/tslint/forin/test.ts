declare const someObject: {};

// good

for (let key in someObject) {
  if (someObject.hasOwnProperty(key)) {
    console.log(key);
  }
}

// bad

for (let key in someObject) {
  console.log(key);
}
