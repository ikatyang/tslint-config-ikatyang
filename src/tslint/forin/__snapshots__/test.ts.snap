// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint 1`] = `
declare const someObject: {};

// good

for (let key in someObject) {
  if (someObject.hasOwnProperty(key)) {
    console.log(key);
  }
}

// bad

for (let key in someObject) {
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  console.log(key);
~~~~~~~~~~~~~~~~~~~
}
~ [for (... in ...) statements must be filtered with an if statement]

`;
